sklearn/tree/tests/test_monotonic_tree.py

```diff
@@ -0,0 +1,331 @@
+import numpy as np
+import pytest
+
+from sklearn.datasets import make_classification, make_regression
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import GradientBoostingRegressor
+from sklearn.tree import DecisionTreeRegressor
+from sklearn.tree.tests.test_tree import REG_TREES, CLF_TREES
+
+
+@pytest.mark.parametrize("seed", range(4))
+@pytest.mark.parametrize("depth_first", (True, False))
+def test_montonic_constraints_classifications(seed, depth_first):
+    n_samples = 1000
+    n_samples_train = 900
```

Comment:
Is it possible to reduce theses sizes while preserving the validity of the test?

Reply:
Not sure, the risk would be to produce mainly trivial or single-split trees.